pkg/ephemeral-disk-utils/utils.go

```diff
@@ -61,19 +60,6 @@ func (om *OwnershipManager) SetFileOwnership(file string) error {
 	if err != nil {
 		return fmt.Errorf("failed to convert GID %s of user %s: %v", owner.Gid, om.user, err)
 	}
-	fileInfo, err := os.Stat(file)
-	if err != nil {
-		return err
-	}
-
-	if stat, ok := fileInfo.Sys().(*syscall.Stat_t); ok {
-		if uid == int(stat.Uid) && gid == int(stat.Gid) {
-			return nil
-		}
-	} else {
-		return fmt.Errorf("failed to convert stat info")
-	}
-
```

Comment:
why do we have to go back to the no-op fake permission setting? that was something i was hoping we could avoid since it has the potential to cover up issues that manifest themselves later on in the vmi start flow.

Reply:
We use the `DefaultOwnershipManager` from 2/3 different contexts. 
1. `virt-handler` - there I am not sure why we would ever not perform the ownership change. Therefore changing this doesn't make sense to me. (And we need to fix the dependency on qemu here)
2. `virt-launcher` (root) - here we should not change the behavior so we don't see any regression. At the same time, I think it makes sense to try get rid of `DefaultOwnershipManager` usage from this context.
3. `virt-launcher` (non-root) - here we should do nothing. It doesn't really help us fail sooner. Without this, you would see on what disk the qemu failed. No less information and also it will be retried as with it. 

Further, I am not sure when we should skip the `chown`(disregarding context). There can be an existing disk in which case it can be owned by `qemu`, part of `qemu` group(or both), or have sufficient `others` permissions. 

The second case is when the disk is not present and we need to create one. There we don't modify the directory at all and can fail on insufficient permissions.

Reply:
i don't want to see no-op behavior like this in our production code.

For example, if i see a function being called in production code (not unit tests) that says it's setting a permission on a file, i don't want to find out that under certain circumstances that function does nothing.